src/globus_sdk/experimental/login_flow_manager/_local_server.py

```diff
@@ -0,0 +1,140 @@
+  </a></div></header>
+
+  <main>
+    <h1>Globus Login Result</h1>
+    <p>
+      $login_result. You may close this tab.
+    <p>
+      $post_login_message
+    </p>
+  </main>
+</body>
+</html>
+"""
+)
+
+
+class LocalServerError(Exception):
+    """
+    Error class for errors raised by the local server when using a
+    LocalServerLoginFlowManager
+    """
+
+
+class RedirectHTTPServer(HTTPServer):
+    """
+    HTTPServer that accepts an html_template to be displayed to the user
+    and uses a Queue to receive an auth_code from its RequestHandler.
+    """
+
+    def __init__(
+        self,
+        server_address: tuple[str, int],
+        handler_class: type[BaseHTTPRequestHandler],
+        html_template: Template,
+    ) -> None:
+        super().__init__(server_address, handler_class)
+
+        self.html_template = html_template
+        self._auth_code_queue: queue.Queue[str | BaseException] = queue.Queue()
+
+    def handle_error(
+        self,
+        request: socket.socket | tuple[bytes, socket.socket],
+        client_address: t.Any,
+    ) -> None:
+        _, excval, _ = sys.exc_info()
+        assert excval is not None
+        self._auth_code_queue.put(excval)
+
+    def return_code(self, code: str | BaseException) -> None:
+        self._auth_code_queue.put_nowait(code)
+
+    def wait_for_code(self) -> str | BaseException:
+        # timeout is set as a workaround for handling control-c interrupt on Windows
+        # https://docs.python.org/3/library/queue.html#queue.Queue.get
+        try:
+            return self._auth_code_queue.get(block=True, timeout=3600)
```

Comment:
I _think_ that this means that `CTRL+C` will not be processed for 1 hour on Windows platforms, which is not a desirable outcome.

There may be other mechanisms that can block without ignoring `CTRL+C` on Windows. I'm specifically considering `threading.Event().wait()` as a possibility for the main thread, so I'll test that on Windows 10 shortly with a few Python versions.

Reply:
Python 3.12 on Windows 10 blocks until the full `timeout` period has passed. _Then_ it processes the `CTRL+C` it received.

Testing alternative implementations now.

Reply:
I spoke with Kurt about this to get some info and learn more -- I wrote some of the early versions of this for the CLI and have definitely touched on this Windows behavior; and I think I've learned that my understanding was not quite correct.

I recommend that we follow the above suggestion to loop, doing short (e.g., 1 second) sleeps.

We discussed and also have the following suggestions:
1. Gate this to Windows only. That way, Linux, macOS, etc get good behavior (a long block on the queue with working Ctrl+C), but Windows users get the best behavior we can offer them. This will require two codepaths where there was only one before, but c'est la vie.
2. Do not trust `time.sleep` to be precise. Calculate a "deadline" and wait until the deadline is reached, rather than doing a fixed count of `time.sleep` calls.

Regarding (2), something like this would work:
```python
deadline = time.time() + 3600
while time.time() < deadline: ...
```